pkg/gardenlet/controller/factory.go

```diff
@@ -128,15 +131,20 @@ func (f *GardenletControllerFactory) Run(ctx context.Context) error {
 		return fmt.Errorf("timed out waiting for Kube caches to sync")
 	}
 
-	secrets, err := garden.ReadGardenSecrets(f.k8sInformers, f.k8sGardenCoreInformers)
-	runtime.Must(err)
-
-	if secret, ok := secrets[common.GardenRoleInternalDomain]; ok {
-		shootList, err := f.k8sGardenCoreInformers.Core().V1beta1().Shoots().Lister().List(labels.Everything())
-		runtime.Must(err)
-		runtime.Must(garden.VerifyInternalDomainSecret(ctx, k8sGardenClient, len(shootList), secret))
+	seedNames := confighelper.SeedNames(f.cfg.SeedConfig, f.k8sGardenCoreInformers.Core().V1beta1().Seeds().Lister(), f.cfg.SeedSelector)
+	if len(seedNames) < 1 {
+		return fmt.Errorf("no seed selected by Gardenlet")
 	}
 
+	// Read Garden secrets from any seed namespace as we assume they are synced accordingly by the Gardener Controller Manager
+	secrets, err := garden.ReadGardenSecrets(
+		ctx,
+		k8sGardenClient.Cache(),
+		f.k8sGardenCoreInformers.Core().V1beta1().Seeds().Lister(),
+		seedpkg.ComputeGardenNamespace(seedNames[0]),
```

Comment:
Why only `seedNames[0]` is taken?

Reply:
It's explained in the comment above:
`// Read Garden secrets from any seed namespace as we assume they are synced accordingly by the Gardener Controller Manager`

Reply:
👍🏻 Makes sense. Maybe you can add another sentence that this might need adaption when it can be better controlled which secrets are replicated for which seeds :)